### PR TITLE
Added method to retrieve documentation contributors.

### PR DESCRIPTION
**Pull request by vanross**
Tests have to be improved after discussing test strategy in next meeting.

### CODE REVIEW

File: src/main/java/net/adoptium/documentationservices/model/Contributor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package net.adoptium.documentationservices.model;
+
+import java.util.Objects;
+
+/**
+ * A person who contributed to the documentation.
+ */
+public class Contributor {
+
+    final private String name;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/documentationservices/model/Contributor.java:10]:
```suggestion
    private final String name;
```

File: src/main/java/net/adoptium/documentationservices/model/Contributor.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    final private String githubAvatar;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/documentationservices/model/Contributor.java:11]:
```suggestion
   private final String githubAvatar;
```

File: src/main/java/net/adoptium/documentationservices/model/Contributor.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    final private String githubProfileURL;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/documentationservices/model/Contributor.java:12]:
```suggestion
    private final String githubProfileURL;
```

File: src/main/java/net/adoptium/documentationservices/services/RepoService.java [hunk 4]
```diff
@@ -60,6 +70,41 @@ public boolean isUpdateAvailable() throws IOException {
...
+            final GHCommitQueryBuilder commitQueryBuilder = repo.queryCommits();
+            PagedIterable<GHCommit> commits = commitQueryBuilder.path(ghContent.getPath()).list();
+            commits.forEach(ghCommit -> {
+                try {
+                    final GHUser author = ghCommit.getAuthor();
+                    if (author != null) {
+                        commitAuthors.add(author);
+                    }
+                } catch (IOException ioe) {
+                    LOG.error("Failed to retrieve author of commit {}.", ghCommit.getSHA1());
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/documentationservices/services/RepoService.java:90]:
```suggestion
                    LOG.error("Failed to retrieve author of commit " + ghCommit.getSHA1(), ioe);
```
by doing so we will log the stack trace

File: src/main/java/net/adoptium/documentationservices/services/RepoService.java [hunk 5]
```diff
@@ -60,6 +70,41 @@ public boolean isUpdateAvailable() throws IOException {
...
+        });
+
+        // convert GHUsers to Contributors
+        final Set<Contributor> contributors = new HashSet<>();
+        commitAuthors.forEach(ghUser -> {
+            try {
+                final Contributor c = new Contributor(ghUser.getName(), ghUser.getAvatarUrl(), GITHUB_WEB_ADDRESS + ghUser.getLogin());
+                contributors.add(c);
+            } catch (IOException ioe) {
+                LOG.error("Failed to retrieve details for GitHub user {}", ghUser.getLogin());
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/documentationservices/services/RepoService.java:102]: Please use log method with exception param to log stack trace.

**Review by hendrikebbers — COMMENTED** [review]
minor change requests

**Comment by vanross** [comment]
Applied all mentioned changes. I'll close this PR and make a new one based on the current state of the main branch to avoid conflicts.

**Pull request closed without merging**